packages/editor/src/components/post-taxonomies/hierarchical-term-selector.js

```diff
@@ -378,194 +356,132 @@ class HierarchicalTermSelector extends Component {
+					required
 				/>
-			),
-			<div
-				className="editor-post-taxonomies__hierarchical-terms-list"
-				key="term-list"
-				tabIndex="0"
-				role="group"
-				aria-label={ groupLabel }
-			>
-				{ this.renderTerms(
-					'' !== filterValue ? filteredTermsTree : availableTermsTree
+				{ !! availableTerms.length && (
+					<TreeSelect
+						label={ parentSelectLabel }
+						noOptionLabel={ noParentOption }
+						onChange={ onChangeFormParent }
+						selectedId={ formParent }
+						tree={ availableTermsTree }
+					/>
 				) }
-			</div>,
-			! loading && hasCreateAction && (
 				<Button
-					key="term-add-button"
-					onClick={ this.onToggleForm }
-					className="editor-post-taxonomies__hierarchical-terms-add"
-					aria-expanded={ showForm }
-					variant="link"
+					variant="secondary"
+					type="submit"
+					className="editor-post-taxonomies__hierarchical-terms-submit"
 				>
-					{ newTermButtonLabel }
+					{ newTermSubmitLabel }
 				</Button>
-			),
-			showForm && (
-				<form onSubmit={ this.onAddTerm } key="hierarchical-terms-form">
-					<label
-						htmlFor={ inputId }
-						className="editor-post-taxonomies__hierarchical-terms-label"
-					>
-						{ newTermLabel }
-					</label>
-					<input
-						type="text"
-						id={ inputId }
-						className="editor-post-taxonomies__hierarchical-terms-input"
-						value={ formName }
-						onChange={ this.onChangeFormName }
-						required
-					/>
-					{ !! availableTerms.length && (
-						<TreeSelect
-							label={ parentSelectLabel }
-							noOptionLabel={ noParentOption }
-							onChange={ this.onChangeFormParent }
-							selectedId={ formParent }
-							tree={ availableTermsTree }
-						/>
-					) }
-					<Button
-						variant="secondary"
-						type="submit"
-						className="editor-post-taxonomies__hierarchical-terms-submit"
-					>
-						{ newTermSubmitLabel }
-					</Button>
-				</form>
-			),
-		];
-	}
+			</form>
+		),
+	];
 }
 
-export default compose( [
-	withSelect( ( select, { slug } ) => {
-		const { getCurrentPost } = select( editorStore );
-		const { getTaxonomy } = select( coreStore );
-		const taxonomy = getTaxonomy( slug );
-		return {
-			hasCreateAction: taxonomy
-				? get(
-						getCurrentPost(),
-						[ '_links', 'wp:action-create-' + taxonomy.rest_base ],
-						false
-				  )
-				: false,
-			hasAssignAction: taxonomy
-				? get(
-						getCurrentPost(),
-						[ '_links', 'wp:action-assign-' + taxonomy.rest_base ],
-						false
-				  )
-				: false,
-			terms: taxonomy
-				? select( editorStore ).getEditedPostAttribute(
-						taxonomy.rest_base
-				  )
-				: [],
-			taxonomy,
-		};
-	} ),
-	withDispatch( ( dispatch ) => ( {
-		onUpdateTerms( terms, restBase ) {
-			dispatch( editorStore ).editPost( { [ restBase ]: terms } );
-		},
-	} ) ),
-	withSpokenMessages,
-	withInstanceId,
-	withFilters( 'editor.PostTaxonomyType' ),
-] )( HierarchicalTermSelector );
+export default withFilters( 'editor.PostTaxonomyType' )(
```

Comment:
There's actually a breaking change with this refactor. Basically in trunk when using the `editor.PostTaxonomyType`  filter, you'd have received all the props added by the withSelect, withDispatch, withInstanceId...  Now these are internal to the component and not available in the hook.

I actually think this was a mistake in the beginning and these props should have never been exposed in that hook. So I think it's a "good" breaking change but we should be careful. Maybe we should check in `wpdirectory.net` if this hook is used and how: potentially contact the plugin authors and also prepare a dev note with this PR.

Reply:
https://wpdirectory.net/search/01FAZ22K1BBF4MWFX99FCY1NH0

The SEO Framework plugin is likely to be affected by this change.